sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/TargetDialog.java

```diff
@@ -8,146 +8,205 @@
+      throw new ParseException("Remote tag cannot be null or empty", 0);
     }
-
-    @Override
-    public String getParameter(String name) {
-        return parameters.getValue(name, true).toString();
+    this.remoteTag = remoteTag.trim();
+  }
+
+  @Override
+  public StringBuilder encodeBody(StringBuilder retval) {
+    if (callId != null) {
+      retval.append(callId.encode());
+      if (localTag != null) {
+        // RFC 4538 uses ";to-tag=" for local tag
+        retval.append(";to-tag=");
+        retval.append(localTag);
+      }
+      if (remoteTag != null) {
+        // RFC 4538 uses ";from-tag=" for remote tag
+        retval.append(";from-tag=");
+        retval.append(remoteTag);
+      }
+      // Encode parameters (if any)
+      if (!parameters.isEmpty()) {
+        retval.append(SEMICOLON).append(parameters.encode());
+      }
     }
-
-    @Override
-    public void setParameter(String name, String value) throws ParseException {
-        parameters.set(name, value);
+    return retval;
+  }
+
+  @Override
+  public Object clone() {
+    TargetDialog retval = (TargetDialog) super.clone();
+    if (callId != null) {
+      retval.callId = (CallIdentifier) callId.clone();
     }
-
-    @Override
-    public Iterator<String> getParameterNames() {
-        return parameters.getNames();
+    retval.localTag = localTag;
+    retval.remoteTag = remoteTag;
+    return retval;
+  }
+
+  @Override
+  public void setValue(String value) throws ParseException {
+    // Not implemented yet, throw an exception to indicate it
+    throw new UnsupportedOperationException("setValue method is not implemented yet");
+  }
+
+  @Override
+  public String getParameter(String name) {
+    return parameters.getValue(name, true).toString();
+  }
+
+  @Override
+  public void setParameter(String name, String value) throws ParseException {
+    parameters.set(name, value);
+  }
+
+  @Override
+  public Iterator<String> getParameterNames() {
+    return parameters.getNames();
+  }
+
+  @Override
+  public void removeParameter(String name) {
+    parameters.delete(name);
+  }
+
+  /**
+   * Parses the header string into Call-Id, local tag, remote tag, and parameters.
+   *
+   * @param body the header string to parse
+   * @throws ParseException if the header string cannot be parsed
+   */
+  public void decodeBody(String body) throws ParseException {
+    int localTagIndex = body.indexOf(";to-tag=");
+    int remoteTagIndex = body.indexOf(";from-tag=");
+    int delimiter = Math.min(localTagIndex != -1 ? localTagIndex : Integer.MAX_VALUE,
+        remoteTagIndex != -1 ? remoteTagIndex : Integer.MAX_VALUE);
+
+    String callIdStr = (delimiter != -1) ? body.substring(0, delimiter) : body;
+    setCallId(callIdStr.trim());
+
+    if (localTagIndex != -1) {
+      int endLocalTag = body.indexOf(';', localTagIndex + 9);
+      localTag = body.substring(localTagIndex + 9, endLocalTag != -1 ? endLocalTag : body.length()).trim();
```

Comment:
Let's use the length of new constant instead

Reply:
ok